tools/sst_dump_tool.cc

```diff
@@ -379,9 +381,18 @@ Status SstFileDumper::ReadSequential(bool print_kv, uint64_t read_num,
     }
 
     if (print_kv) {
-      fprintf(stdout, "%s => %s\n",
-          ikey.DebugString(output_hex_).c_str(),
-          value.ToString(output_hex_).c_str());
+      std::string key_str = ikey.DebugString(output_hex_);
+      std::string val_str = value.ToString(output_hex_);
+
+      if (output_safe_) {
+        std::replace(key_str.begin(), key_str.end(), '\n', '\0');
+        std::replace(val_str.begin(), val_str.end(), '\n', '\0');
```

Comment:
Just make sure you are doing what you are expected. Are you going to turn new line into a null terminator, so that only the first line is printed out?

Reply:
>so that only the first line is printed out

not sure that I understand you, my idea was to replace LF (`\n`) with NUL (`\0`) byte to avoid breaking parsing, so for example if value contains LF `sst_dump` will break the current line and one may interpret next line as a new key-value pair.

Reply:
In this case, you may use `--output_hex` to examine the hex.

Reply:
Kind of, but `--output-hex` has too huge overhead, i.e. you cannot use `sst_dump | fgrep | awk` in this case, since it will convert even ASCII

Reply:
If we want to claim a print safe version, a better idea is to replace '\n' to be '\\n'. Similarly, there are other characters need to be escaped too. Like '\0', '\t', '\r', etc. If you can bring a mostly complete escaping solution, we can merge it.

Reply:
If you want to search for a key "abcd\nefg" in which \\n' is LF, it is possible to convert the key to hex first and do `grep` and `awk`. It's much less intuitive though.

Reply:
>If we want to claim a print safe version, a better idea is to replace '\n' to be '\n'. Similarly, there are other characters need to be escaped too. Like '\0', '\t', '\r', etc.

Indeed, fixed.

>If you want to search for a key "abcd\nefg" in which \n' is LF, it is possible to convert the key to hex first and do grep and awk. It's much less intuitive though.

Sure I can do this, but output in hex is too much overwhelming for my use case